examples/elaboratePoint2KalmanFilter.cpp

```diff
@@ -55,17 +55,21 @@ int main() {
 
   // Create new state variable
   Symbol x0('x',0);
-  ordering->insert(x0, 0);
+  ordering->push_back(x0);
 
   // Initialize state x0 (2D point) at origin by adding a prior factor, i.e., Bayes net P(x0)
   // This is equivalent to x_0 and P_0
   Point2 x_initial(0,0);
-  SharedDiagonal P_initial = noiseModel::Diagonal::Sigmas((Vec(2) << 0.1, 0.1));
-  PriorFactor<Point2> factor1(x0, x_initial, P_initial);
+  SharedDiagonal P_initial = noiseModel::Diagonal::Sigmas((gtsam::Vector2() << 0.1, 0.1).finished());
+
+   // Create a JacobianFactor directly - this represents the prior constraint on x0
+   JacobianFactor::shared_ptr factor1(
+	new JacobianFactor(x0, P_initial->R(), Vector::Zero(2),
+	  noiseModel::Unit::Create(2)));
+  
   // Linearize the factor and add it to the linear factor graph
   linearizationPoints.insert(x0, x_initial);
-  linearFactorGraph->push_back(factor1.linearize(linearizationPoints, *ordering));
-
+  linearFactorGraph->push_back(factor1);
```

Comment:
Try to use emplace_shared if you’re creating the shared pointer.

Reply:
-> emplace_shared<JacobianFactor>(args)
OR
-> add(args)

so make_shared above is no longer needed.